addons/website_sale/static/src/js/variant_mixin.js

```diff
@@ -57,6 +57,20 @@ VariantMixin._onChangeCombination = function (ev, $parent, combination) {
         contactUsButton.removeClass('d-flex').addClass('d-none');
         product_unavailable.removeClass('d-flex').addClass('d-none')
     }
+
+    // Dynamically render extra fields
+    let $extras = $('#product-extra-fields');
```

Comment:
the id `#product-extra-fields` is defined in this PR to be used as selector which is something to avoid as views templates are not updated automatically, which means your js code will never find this selector for old clients, that's why in case it's necessary for the fix to use a selector we must manage that with what we already have.
Also we need to check the result of the query is not falsy before using it or the code will crash